external/CMakeLists.txt

```diff
@@ -16,18 +16,18 @@ file(GLOB_RECURSE external_src
   "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp"
   )
 add_library(external ${external_src})
-target_include_directories(external PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+target_include_directories(external SYSTEM PUBLIC  ${CMAKE_CURRENT_SOURCE_DIR}/include)
```

Comment:
What does `SYSTEM` do?

Reply:
makes it a system header so we dont get warnings from it